text/0000-machine-readable-tests-output.md

```diff
@@ -0,0 +1,146 @@
+- Feature Name: machine\_readable\_tests\_output
+  - Start Date: 2015-09-17
+  - RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Replace current test output with machine-readable one and add thin compatibility
+layer on top of that.
+
+# Motivation
+
+Currently when testing Rust provide only human-readable output which is nice
+in most cases as we are humans or other humanoids. But from time to time we need
+to feed machine with results of our tests (i.e. some kind of CI that will report
+which tests failed or something like that) and there is no way to do that in
+"civilised" way. We need to parse non-machine-readable output which isn't nice.
+
+# Detailed design
+
+Replace current, human-readable, output with JSON-based output based on
+[TAP-J][tap-j]. For compatibility with existing work flow there should be added
+built-in parser for that protocol which will output in current format.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119][rfc2119].
+
+## Protocol description
+
+Output of test MUST be stream of JSON objects that can be parsed by external
+tools.
+
+### Structure
+
+Output is stream of lines containing JSON objects separated by new line. Each
```

Comment:
It could also make usage of `\1E` ASCII code which refers to "Record Separator". This is still under consideration, but 1 record per line is simple enough.